pkg/kubelet/container_gc.go

```diff
@@ -157,6 +170,18 @@ func (cgc *realContainerGC) GarbageCollect() error {
 		}
 	}
 
+	// Remove dead symlinks - should only happen on upgrade
+	// from a k8s version without proper log symlink cleanup
+	logSymlinks, _ := filepath.Glob(path.Join(cgc.containerLogsDir, "*.log"))
+	for _, logSymlink := range logSymlinks {
+		if _, err = os.Stat(logSymlink); os.IsNotExist(err) {
```

Comment:
I didn't write test code, I just used the standard 'stat' command line tool, as per my comment above (without the -L option to follow links).